packages/builtin/array.pony

```diff
@@ -1,6 +1,78 @@
 class Array[A] is Seq[A]
   """
   Contiguous, resizable memory to store elements of type A.
+
+  ## Usage
+  Creating an Array of String.
+  ```pony
+    let array: Array[String] = ["dog"; "cat"; "wombat"]
+  ```
+
+  Accessing elements can be done via the `apply(i: USize): this->A ?` method.
+  The provided index might be out of bounds so `apply` (like many other Array's
+  methods) is partial and has to be called within a try-catch block or inside
+  another partial method.
+  ```pony
+    let array: Array[String] = ["dog"; "cat"; "wombat"]
+    let is_second_element_wobat = try
+      // indexes start from 0, so 1 is the second element
+      array(1)? == "wombat"
+    else
+      false
+    end
+  ```
+
+  Adding and removing elements to and from the end of the Array can be done via
+  `push` and `pop` methods, which add and remove elements to the end of the
+  Array.
+
+  Modifying the Array can be done via `update`, `insert` and `delete` methods
+  which alter the Array at an arbitrary index, moving elements left (when
+  deleting) or right (when inserting) as necessary.
+
+  Iterating over can be done using the `values` method.
```

Comment:
Something seems missing here. Suggestion: `Iterating over the elements of an Array`.

This particular feature of Array definitely needs some more explanation, as it returns aliases on the elemens as the array sees them (taking viewpoint adaptation into account), which is a complicated topic for some cases. But this can be part of another PR.